README.md

```diff
@@ -2,7 +2,140 @@
+```julia
+generator_ids = [111, 222, 333]
+datetimes = DateTime(2017, 12, 15):Hour(1):DateTime(2017, 12, 15, 23)
+gen_time_series = GeneratorTimeSeries(;
+    initial_generation = KeyedArray(fill(100.0, length(generator_ids)); generator_ids),
+    offer_curve = KeyedArray(fill([(1.0, 100.0)], length(generator_ids), length(datetimes)); generator_ids, datetimes)
+    ...
+)
+```
+The `GeneratorTimeSeries` type includes fields for additional time series such as generation limits, ramp limits and ancillary service offers, which means these features can be included in an optimisation problem (see [`GeneratorTimeSeries`](https://invenia.github.io/FullNetworkSystems.jl/stable/#FullNetworkSystems.GeneratorTimeSeries) for details of all the time series fields).
+`GeneratorStatus` types also contain time series data, specifically associated with the status of the generator (e.g. whether it is on or off, how long it has been on or off).
+This is useful for including factors such as ramp rates in an optimisation problem, as the status of a generator limits how much it can ramp up or down in a given hour.
+
+The `SystemDA` type includes fields for additional supply (`increment`) and demand (`decrement`, `price_sensitive_loads`) bids for the purpose of modelling virtual participation in electricity markets ([introduction to virtual bidding](https://hepg.hks.harvard.edu/publications/virtual-bidding-and-electricity-market-design)).
+The bid data is expected to be a `KeyedArray` where the dimensions are `ids x datetimes` and the fields are vectors of price-volume pairs, as in the offer curves.
+```julia
+bid_ids = ["123", "456", "789"] # unique identifers
+datetimes = DateTime(2017, 12, 15):Hour(1):DateTime(2017, 12, 15, 23)
+increments = KeyedArray(
+    fill([(0.1, 10.0)], length(generator_ids), length(datetimes)); bid_ids, datetimes
+)
+```
+
+## Grid Matrices
+
+This package also provides functions to calculate the Power Transfer Distribution Factor (PTDF) and Line Outage Distribution Factor (LODF) matrices for a `System` of branches and buses.
+These matrices are useful for modelling branch flow constraints under different contingency scenarios.
+The `System` types have fields to store these matrices.
+
+## Example Modelling Problem
+
+The following is an example of how the data stored in a `SystemRT` object can be used to build a JuMP model.
+The objective of the model is to solve a simple energy balance problem, where a set of loads (demands) need to be met by generation (supply), for the minimum possible cost.
+Assume we have built all the components of the system described in the previous sections so that we can construct an instance of a system (see [`SystemRT`](https://invenia.github.io/FullNetworkSystems.jl/stable/#FullNetworkSystems.SystemRT) for specific details of the components in a `SystemRT`).
+
+```julia
+using AxisKeys
+using FullNetworkSystems
+using HiGHS
+using JuMP
+
+system = SystemRT(
```

Comment:
maybe just add a comment here on the simplifying assumption of this example

Reply:
Added a comment, does it clarify things?